### DISCUSSION TITLE
Using ONNX as an intermediary

### DISCUSSION THREAD

**Discussion by modanesh** [onnx/onnx#4755] [Q&A]
I'm trying to load a `pth` model in Go. Can ONNX be used as an intermediary to convert a model from Pytorch's `pth` to `gob` in GoLang?

**Comment by xadupre** [comment]
I don't know too much about go. Once your model is exported in onnx, it can be seen as a long binary string. This package seems able to read it: https://pkg.go.dev/github.com/tynrol/onnx-go.

**Reply by modanesh** [reply]
Thanks @xadupre 
I tried the [ONNX-GO](https://github.com/owulveryck/onnx-go) initially. From my understanding, ONNX-GO loads the model into a computation graph and then runs it. In the ONNX-GO's repo, it's recommended to use [Gorgonia](https://github.com/gorgonia/gorgonia) as the backend, and that's what I did. However, in order to run the model, it needs to already have all the required operators of the model implemented. And for my case, there was a missing operator. I created an issue in the ONNX-GO's repo (https://github.com/owulveryck/onnx-go/issues/203), but no reply. That's when I went for plan B, which was to use ONNX as an intermediary to convert a PyTorch's `pth` model into a GoLang's `gob`, hence the question above.

**Reply by BEEugene** [reply]
There is also another project, have you tried that one? https://github.com/AdvancedClimateSystems/gonnx/tree/main
